Tests/StreamChatTests/APIClient/Endpoints/Payloads/ChannelListPayload_Tests.swift

```diff
@@ -30,22 +30,36 @@ final class ChannelListPayload_Tests: XCTestCase {
             }
         }
     }
-    
-    func test_hugeChannelListQuery_save() throws {
+
+    func saveChannelListPayload(_ payload: ChannelListPayload, database: DatabaseContainer_Spy, timeout: TimeInterval = 10) {
+        let writeCompleted = expectation(description: "DB write complete")
+        database.write({ session in
+            session.saveChannelList(payload: payload, query: .init(filter: .containMembers(userIds: [.unique])))
+        }, completion: { error in
+            if let error = error {
+                XCTFail("DB write error: \(error)")
+            }
+            writeCompleted.fulfill()
+        })
+        wait(for: [writeCompleted], timeout: timeout)
+    }
+
+    func test_hugeChannelListQuery_save_uncached() throws {
         let decodedPayload = createHugeChannelList()
-        
         measure {
             let databaseContainer = DatabaseContainer_Spy()
-            let writeCompleted = expectation(description: "DB write complete")
-            databaseContainer.write({ session in
-                try session.saveChannelList(payload: decodedPayload, query: .init(filter: .containMembers(userIds: [.unique])))
-            }, completion: { error in
-                if let error = error {
-                    XCTFail("DB write error: \(error)")
-                }
-                writeCompleted.fulfill()
-            })
-            wait(for: [writeCompleted], timeout: 10)
+            saveChannelListPayload(decodedPayload, database: databaseContainer)
+        }
+    }
+
+    func test_hugeChannelListQuery_save_cached() throws {
+        let decodedPayload = createHugeChannelList()
+        let databaseContainer = DatabaseContainer_Spy()
```

Comment:
`databaseContainer` creation should happen in `measure` block (as it's in the uncached test) since DB creation makes sure we're writing the payload on an empty DB

We can introduce another set of tests where DB creation happens outside `measure` block and this'll measure payload saving when DB already has the data

Reply:
This is exactly what we don't want to do. In a pre-warm test, what you want is to have the DB filled, and measure how much it takes to save the payload when the data is already there.

Reply:
The other test is measuring how much time it takes to save the payload _when DB is empty_. To be able to compare, this test should also measure that. As I said, we can have another set of tests where we measure this on a pre-filled DB.

Pre-warm doesn't only benefit us when DB is filled, it also benefits when DB is empty since # of fetches are greatly reduced

Reply:
I guess you are misled by the test name. What I mean by cached is with the DB filled.

I both cases Pre-warming is enabled.